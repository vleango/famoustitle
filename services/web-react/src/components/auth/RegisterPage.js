import React, { Component, Fragment } from 'react';
import { connect } from 'react-redux';
import { Button } from 'reactstrap';
import { Form, FormGroup, Input } from 'reactstrap';

import { startRegister } from '../../actions/auth';
import Header from '../shared/headers/Header';

export class RegisterPage extends Component {

    constructor(props) {
        super(props);
        this.state = {
            first_name: "",
            last_name: "",
            email: "",
            password: "",
            password_confirmation: "",
            submitting: false,
            errorMsg: ""
        };
    }

    onInputChange = (e) => {
        const field = e.target.name;
        const value = e.target.value;
        this.setState(() => ({ [field]: value }));
    };

    onSubmitRegister = async (e) => {
        e && e.preventDefault();
        const { first_name, last_name, email, password, password_confirmation } = this.state;
        if(first_name === "" || last_name === "" || email === "" || password === "" || password_confirmation === "") {
            this.setState({ errorMsg: "required field is missing" });
            return;
        }

        if(password.length < 6) {
            this.setState({ errorMsg: "password is less than 6 characters" });
            return;
        }

        if(password !== password_confirmation) {
            this.setState({ errorMsg: "passwords does not match" });
            return;
        }

        try {
            this.setState({ submitting: true, errorMsg: "" });
            await this.props.startRegister({
                user: {
                    first_name: this.state.first_name,
                    last_name: this.state.last_name,
                    email: this.state.email
                },
                password: this.state.password,
                password_confirmation: this.state.password_confirmation
            });
            this.props.history.push('/');
        }
        catch (err) {
            this.setState({ submitting: false, errorMsg: "something went wrong..." });
        }
    };

    render() {
        return (
            <Fragment>
                <Header resourceTitle="Register" />
                <div className="container">
                    <Form onSubmit={this.onSubmitRegister} autoComplete="off">
                        <FormGroup>
                            <Input type="text"
                                   name="first_name"
                                   value={this.state.first_name}
                                   placeholder="First name"
                                   onChange={this.onInputChange} />
                            <Input type="text"
                                   name="last_name"
                                   value={this.state.last_name}
                                   placeholder="Last name"
                                   onChange={this.onInputChange} />
                            <Input type="email"
                                   name="email"
                                   value={this.state.email}
                                   placeholder="Email"
                                   onChange={this.onInputChange} />
                        </FormGroup>
                        <FormGroup>
                            <Input type="password"
                                   name="password"
                                   value={this.state.password}
                                   placeholder="password"
                                   onChange={this.onInputChange} />
                            <Input type="password"
                                   name="password_confirmation"
                                   value={this.state.password_confirmation}
                                   placeholder="password confirmation"
                                   onChange={this.onInputChange} />
                        </FormGroup>

                        { this.state.errorMsg && <p>{this.state.errorMsg}</p> }

                        <div className="clearfix">
                            <Button color="primary float-right" disabled={this.state.submitting} size="lg">Register</Button>
                        </div>
                    </Form>
                </div>
            </Fragment>
        );
    }

}

const mapDispatchToProps = (dispatch) => ({
    startRegister: async (data) => await dispatch(startRegister(data))
});

export default connect(null, mapDispatchToProps)(RegisterPage);
