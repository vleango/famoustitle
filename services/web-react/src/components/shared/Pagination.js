import React, { Fragment } from 'react';
import { Link } from 'react-router-dom';
import { forEach, range } from 'lodash';
import { Pagination, PaginationItem, PaginationLink } from 'reactstrap';

import './css/pagination.css'

export default (props) => {
  const {currentPage, totalPages} = props;
  if(currentPage === 0 || currentPage > totalPages) {
    return (
      <Fragment>{''}</Fragment>
    );
  }

  const pageBoxes = 5;
  let selectedPagePosition = currentPage % pageBoxes;

  if(selectedPagePosition === 0) {
    selectedPagePosition = pageBoxes;
  }

  let paginationItems = [];
  forEach(range(1, pageBoxes + 1), (index) => {
    let pageNumber = currentPage - selectedPagePosition + index;
    if (pageNumber <= totalPages) {
      paginationItems.push(
        <PaginationItem key={"pagination-item-" + index} active={ selectedPagePosition === index }>
          <PaginationLink tag={Link} to={`/?currentPage=${pageNumber}`}>
            { pageNumber }
          </PaginationLink>
        </PaginationItem>
      );
    }
  });

  const nextPageSet = currentPage + (pageBoxes - selectedPagePosition) + 1;

  return (
    <Pagination>
      <PaginationItem hidden={currentPage <= pageBoxes }>
        <PaginationLink previous tag={Link} to={`/?currentPage=${currentPage - selectedPagePosition}`} />
      </PaginationItem>
      { paginationItems }
      <PaginationItem hidden={nextPageSet >= totalPages}>
        <PaginationLink next tag={Link} to={`/?currentPage=${nextPageSet}`} />
      </PaginationItem>
    </Pagination>
  );
}
