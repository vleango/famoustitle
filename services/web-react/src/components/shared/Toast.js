import React  from 'react';
import { ToastContainer, toast } from 'react-toastify';
import 'react-toastify/dist/ReactToastify.css';

export const toastInProgress = (message) => {
    return toast(message, { autoClose: false });
};

export const toastSuccess = (message, toastID = null) => {
    if(toastID !== null) {
        toast.update(toastID, { render: message, type: toast.TYPE.SUCCESS, autoClose: 3000 });
    } else {
        toast.success(message);
    }
};

export const toastFail = (message, toastID = null) => {
    if(toastID !== null) {
        toast.update(toastID, { render: message, type: toast.TYPE.ERROR, autoClose: 3000 });
    } else {
        toast.error(message);
    }
};

export default () => {

    return (
        <ToastContainer position="top-center"
                        autoClose={5000}
                        hideProgressBar
                        newestOnTop={false}
                        closeOnClick
                        rtl={false}
                        pauseOnVisibilityChange
                        draggable
                        pauseOnHover />
    );
}
