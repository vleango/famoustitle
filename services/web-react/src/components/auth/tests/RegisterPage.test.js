import React from 'react';
import { shallow } from 'enzyme';

import { RegisterPage } from '../RegisterPage';

import { toastInProgress, toastSuccess, toastFail } from '../../shared/Toast';

jest.mock('../../shared/Toast', () => ({
    toastInProgress: jest.fn((message, id) => { return 1 }),
    toastSuccess: jest.fn((message, id) => { return 2 }),
    toastFail: jest.fn((message, id) => { return 3 })
}));

beforeEach(() => {
    jest.clearAllMocks();
});

describe('Components', () => {
    describe('Auth', () => {
        describe('RegisterPage', () => {

            let wrapper = shallow(<RegisterPage />);

            describe('Snapshot', () => {
                it('should correctly render RegisterPage', () => {
                    expect(wrapper).toMatchSnapshot();
                });
            });

            describe('onSubmitRegister', () => {
                describe('first_name is blank', () => {
                    it('sets errorMsg', () => {
                        wrapper.setState({ first_name: "", last_name: "Leang", email: "tha@test.com", password: "hogehoge", password_confirmation: "hogehoge" });
                        wrapper.instance().onSubmitRegister();
                        expect(wrapper.state('errorMsg')).toBe("required field is missing");
                    });
                });

                describe('last_name is blank', () => {
                    it('sets errorMsg', () => {
                        wrapper.setState({ first_name: "Tha", last_name: "", email: "tha@test.com", password: "hogehoge", password_confirmation: "hogehoge" });
                        wrapper.instance().onSubmitRegister();
                        expect(wrapper.state('errorMsg')).toBe("required field is missing");
                    });
                });

                describe('email is blank', () => {
                    it('sets errorMsg', () => {
                        wrapper.setState({ first_name: "Tha", last_name: "Leang", email: "", password: "hogehoge", password_confirmation: "hogehoge" });
                        wrapper.instance().onSubmitRegister();
                        expect(wrapper.state('errorMsg')).toBe("required field is missing");
                    });
                });

                describe('password is blank', () => {
                    it('sets errorMsg', () => {
                        wrapper.setState({ first_name: "Tha", last_name: "Leang", email: "tha@test.com", password: "", password_confirmation: "hogehoge" });
                        wrapper.instance().onSubmitRegister();
                        expect(wrapper.state('errorMsg')).toBe("required field is missing");
                    });
                });

                describe('password_confirmation is blank', () => {
                    it('sets errorMsg', () => {
                        wrapper.setState({ first_name: "Tha", last_name: "Leang", email: "tha@test.com", password: "hogehoge", password_confirmation: "" });
                        wrapper.instance().onSubmitRegister();
                        expect(wrapper.state('errorMsg')).toBe("required field is missing");
                    });
                });

                describe('passwords do not match', () => {
                    it('sets errorMsg', () => {
                        wrapper.setState({ first_name: "Tha", last_name: "Leang", email: "tha@test.com", password: "hogehoge", password_confirmation: "piyopiyo" });
                        wrapper.instance().onSubmitRegister();
                        expect(wrapper.state('errorMsg')).toBe("passwords does not match");
                    });
                });

                describe('password less than 6 characters', () => {
                    it('sets errorMsg', () => {
                        wrapper.setState({ first_name: "Tha", last_name: "Leang", email: "tha@test.com", password: "12345", password_confirmation: "12345" });
                        wrapper.instance().onSubmitRegister();
                        expect(wrapper.state('errorMsg')).toBe("password is less than 6 characters");
                    });
                });

                describe('first_name, last_name, email, password, and password_confirmation are given', () => {
                   it('Success', async () => {
                       const startRegister = jest.fn(() => Promise.resolve());
                       let wrapper = shallow(<RegisterPage startRegister={startRegister} history={[]} />);

                       wrapper.setState({ first_name: "Tha", last_name: "Leang", email: "tha@test.com", password: "123456", password_confirmation: "123456" });
                       await wrapper.instance().onSubmitRegister();
                       expect(wrapper.state('submitting')).toBe(true);
                       expect(wrapper.state('errorMsg')).toBe("");
                       expect(toastInProgress).toHaveBeenCalledWith("Creating your account...");
                       expect(toastSuccess).toHaveBeenCalledWith("Success!", 1);
                       expect(toastFail).not.toHaveBeenCalled();
                   });

                   it('Fail', async () => {
                       const startRegister = jest.fn(() => Promise.reject());
                       let wrapper = shallow(<RegisterPage startRegister={startRegister} history={[]} />);

                       wrapper.setState({ first_name: "Tha", last_name: "Leang", email: "tha@test.com", password: "123456", password_confirmation: "123456" });
                       await wrapper.instance().onSubmitRegister();
                       expect(wrapper.state('submitting')).toBe(false);
                       expect(wrapper.state('errorMsg')).toBe("something went wrong...");
                       expect(toastInProgress).toHaveBeenCalledWith("Creating your account...");
                       expect(toastSuccess).not.toHaveBeenCalled();
                       expect(toastFail).toHaveBeenCalledWith("something went wrong...", 1);
                   });
                });
            });

        });

    });
});
