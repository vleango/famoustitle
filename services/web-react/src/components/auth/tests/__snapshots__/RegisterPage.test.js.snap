// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components Auth RegisterPage Snapshot should correctly render RegisterPage 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RegisterPage />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <div
        className="container"
>
        <Form
                autoComplete="off"
                onSubmit={[Function]}
                tag="form"
        >
                <FormGroup
                        tag="div"
                >
                        <Input
                                name="first_name"
                                onChange={[Function]}
                                placeholder="First name"
                                type="text"
                                value=""
                        />
                        <Input
                                name="last_name"
                                onChange={[Function]}
                                placeholder="Last name"
                                type="text"
                                value=""
                        />
                        <Input
                                name="email"
                                onChange={[Function]}
                                placeholder="Email"
                                type="email"
                                value=""
                        />
                </FormGroup>
                <FormGroup
                        tag="div"
                >
                        <Input
                                name="password"
                                onChange={[Function]}
                                placeholder="password"
                                type="password"
                                value=""
                        />
                        <Input
                                name="password_confirmation"
                                onChange={[Function]}
                                placeholder="password confirmation"
                                type="password"
                                value=""
                        />
                </FormGroup>
                
                <div
                        className="clearfix"
                >
                        <Button
                                color="primary float-right"
                                disabled={false}
                                size="lg"
                                tag="button"
                        >
                                Register
                        </Button>
                </div>
        </Form>
</div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Form
          autoComplete="off"
          onSubmit={[Function]}
          tag="form"
>
          <FormGroup
                    tag="div"
          >
                    <Input
                              name="first_name"
                              onChange={[Function]}
                              placeholder="First name"
                              type="text"
                              value=""
                    />
                    <Input
                              name="last_name"
                              onChange={[Function]}
                              placeholder="Last name"
                              type="text"
                              value=""
                    />
                    <Input
                              name="email"
                              onChange={[Function]}
                              placeholder="Email"
                              type="email"
                              value=""
                    />
          </FormGroup>
          <FormGroup
                    tag="div"
          >
                    <Input
                              name="password"
                              onChange={[Function]}
                              placeholder="password"
                              type="password"
                              value=""
                    />
                    <Input
                              name="password_confirmation"
                              onChange={[Function]}
                              placeholder="password confirmation"
                              type="password"
                              value=""
                    />
          </FormGroup>
          
          <div
                    className="clearfix"
          >
                    <Button
                              color="primary float-right"
                              disabled={false}
                              size="lg"
                              tag="button"
                    >
                              Register
                    </Button>
          </div>
</Form>,
        "className": "container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "autoComplete": "off",
          "children": Array [
            <FormGroup
              tag="div"
>
              <Input
                            name="first_name"
                            onChange={[Function]}
                            placeholder="First name"
                            type="text"
                            value=""
              />
              <Input
                            name="last_name"
                            onChange={[Function]}
                            placeholder="Last name"
                            type="text"
                            value=""
              />
              <Input
                            name="email"
                            onChange={[Function]}
                            placeholder="Email"
                            type="email"
                            value=""
              />
</FormGroup>,
            <FormGroup
              tag="div"
>
              <Input
                            name="password"
                            onChange={[Function]}
                            placeholder="password"
                            type="password"
                            value=""
              />
              <Input
                            name="password_confirmation"
                            onChange={[Function]}
                            placeholder="password confirmation"
                            type="password"
                            value=""
              />
</FormGroup>,
            "",
            <div
              className="clearfix"
>
              <Button
                            color="primary float-right"
                            disabled={false}
                            size="lg"
                            tag="button"
              >
                            Register
              </Button>
</div>,
          ],
          "onSubmit": [Function],
          "tag": "form",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <Input
                  name="first_name"
                  onChange={[Function]}
                  placeholder="First name"
                  type="text"
                  value=""
/>,
                <Input
                  name="last_name"
                  onChange={[Function]}
                  placeholder="Last name"
                  type="text"
                  value=""
/>,
                <Input
                  name="email"
                  onChange={[Function]}
                  placeholder="Email"
                  type="email"
                  value=""
/>,
              ],
              "tag": "div",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "name": "first_name",
                  "onChange": [Function],
                  "placeholder": "First name",
                  "type": "text",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "name": "last_name",
                  "onChange": [Function],
                  "placeholder": "Last name",
                  "type": "text",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "name": "email",
                  "onChange": [Function],
                  "placeholder": "Email",
                  "type": "email",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <Input
                  name="password"
                  onChange={[Function]}
                  placeholder="password"
                  type="password"
                  value=""
/>,
                <Input
                  name="password_confirmation"
                  onChange={[Function]}
                  placeholder="password confirmation"
                  type="password"
                  value=""
/>,
              ],
              "tag": "div",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "name": "password",
                  "onChange": [Function],
                  "placeholder": "password",
                  "type": "password",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "name": "password_confirmation",
                  "onChange": [Function],
                  "placeholder": "password confirmation",
                  "type": "password",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Button
                color="primary float-right"
                disabled={false}
                size="lg"
                tag="button"
>
                Register
</Button>,
              "className": "clearfix",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Register",
                "color": "primary float-right",
                "disabled": false,
                "size": "lg",
                "tag": "button",
              },
              "ref": null,
              "rendered": "Register",
              "type": [Function],
            },
            "type": "div",
          },
        ],
        "type": [Function],
      },
      "type": "div",
    },
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <div
          className="container"
>
          <Form
                    autoComplete="off"
                    onSubmit={[Function]}
                    tag="form"
          >
                    <FormGroup
                              tag="div"
                    >
                              <Input
                                        name="first_name"
                                        onChange={[Function]}
                                        placeholder="First name"
                                        type="text"
                                        value=""
                              />
                              <Input
                                        name="last_name"
                                        onChange={[Function]}
                                        placeholder="Last name"
                                        type="text"
                                        value=""
                              />
                              <Input
                                        name="email"
                                        onChange={[Function]}
                                        placeholder="Email"
                                        type="email"
                                        value=""
                              />
                    </FormGroup>
                    <FormGroup
                              tag="div"
                    >
                              <Input
                                        name="password"
                                        onChange={[Function]}
                                        placeholder="password"
                                        type="password"
                                        value=""
                              />
                              <Input
                                        name="password_confirmation"
                                        onChange={[Function]}
                                        placeholder="password confirmation"
                                        type="password"
                                        value=""
                              />
                    </FormGroup>
                    
                    <div
                              className="clearfix"
                    >
                              <Button
                                        color="primary float-right"
                                        disabled={false}
                                        size="lg"
                                        tag="button"
                              >
                                        Register
                              </Button>
                    </div>
          </Form>
</div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Form
            autoComplete="off"
            onSubmit={[Function]}
            tag="form"
>
            <FormGroup
                        tag="div"
            >
                        <Input
                                    name="first_name"
                                    onChange={[Function]}
                                    placeholder="First name"
                                    type="text"
                                    value=""
                        />
                        <Input
                                    name="last_name"
                                    onChange={[Function]}
                                    placeholder="Last name"
                                    type="text"
                                    value=""
                        />
                        <Input
                                    name="email"
                                    onChange={[Function]}
                                    placeholder="Email"
                                    type="email"
                                    value=""
                        />
            </FormGroup>
            <FormGroup
                        tag="div"
            >
                        <Input
                                    name="password"
                                    onChange={[Function]}
                                    placeholder="password"
                                    type="password"
                                    value=""
                        />
                        <Input
                                    name="password_confirmation"
                                    onChange={[Function]}
                                    placeholder="password confirmation"
                                    type="password"
                                    value=""
                        />
            </FormGroup>
            
            <div
                        className="clearfix"
            >
                        <Button
                                    color="primary float-right"
                                    disabled={false}
                                    size="lg"
                                    tag="button"
                        >
                                    Register
                        </Button>
            </div>
</Form>,
          "className": "container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "autoComplete": "off",
            "children": Array [
              <FormGroup
                tag="div"
>
                <Input
                                name="first_name"
                                onChange={[Function]}
                                placeholder="First name"
                                type="text"
                                value=""
                />
                <Input
                                name="last_name"
                                onChange={[Function]}
                                placeholder="Last name"
                                type="text"
                                value=""
                />
                <Input
                                name="email"
                                onChange={[Function]}
                                placeholder="Email"
                                type="email"
                                value=""
                />
</FormGroup>,
              <FormGroup
                tag="div"
>
                <Input
                                name="password"
                                onChange={[Function]}
                                placeholder="password"
                                type="password"
                                value=""
                />
                <Input
                                name="password_confirmation"
                                onChange={[Function]}
                                placeholder="password confirmation"
                                type="password"
                                value=""
                />
</FormGroup>,
              "",
              <div
                className="clearfix"
>
                <Button
                                color="primary float-right"
                                disabled={false}
                                size="lg"
                                tag="button"
                >
                                Register
                </Button>
</div>,
            ],
            "onSubmit": [Function],
            "tag": "form",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  <Input
                    name="first_name"
                    onChange={[Function]}
                    placeholder="First name"
                    type="text"
                    value=""
/>,
                  <Input
                    name="last_name"
                    onChange={[Function]}
                    placeholder="Last name"
                    type="text"
                    value=""
/>,
                  <Input
                    name="email"
                    onChange={[Function]}
                    placeholder="Email"
                    type="email"
                    value=""
/>,
                ],
                "tag": "div",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "name": "first_name",
                    "onChange": [Function],
                    "placeholder": "First name",
                    "type": "text",
                    "value": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "name": "last_name",
                    "onChange": [Function],
                    "placeholder": "Last name",
                    "type": "text",
                    "value": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "name": "email",
                    "onChange": [Function],
                    "placeholder": "Email",
                    "type": "email",
                    "value": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  <Input
                    name="password"
                    onChange={[Function]}
                    placeholder="password"
                    type="password"
                    value=""
/>,
                  <Input
                    name="password_confirmation"
                    onChange={[Function]}
                    placeholder="password confirmation"
                    type="password"
                    value=""
/>,
                ],
                "tag": "div",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "name": "password",
                    "onChange": [Function],
                    "placeholder": "password",
                    "type": "password",
                    "value": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "name": "password_confirmation",
                    "onChange": [Function],
                    "placeholder": "password confirmation",
                    "type": "password",
                    "value": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            "",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Button
                  color="primary float-right"
                  disabled={false}
                  size="lg"
                  tag="button"
>
                  Register
</Button>,
                "className": "clearfix",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Register",
                  "color": "primary float-right",
                  "disabled": false,
                  "size": "lg",
                  "tag": "button",
                },
                "ref": null,
                "rendered": "Register",
                "type": [Function],
              },
              "type": "div",
            },
          ],
          "type": [Function],
        },
        "type": "div",
      },
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
